urlparse4/cgurl.pyx

```diff
@@ -134,36 +162,45 @@ class SplitResultNamedTuple(tuple):
             elif prop == "fragment":
                 return self[4]
             elif prop == "port":
+                """
+                TODO:
+                Port can go beyond 0
+                """
                 if parsed.port.len > 0:
                     port = int(slice_component(url, parsed.port))
                     if port <= 65535:
                         return port
 
             elif prop == "username":
+                if decoded:
+                    return slice_component(url, parsed.username).decode('utf-8') or None
                 return slice_component(url, parsed.username) or None
             elif prop == "password":
+                if decoded:
+                    return slice_component(url, parsed.password).decode('utf-8') or None
                 return slice_component(url, parsed.password) or None
             elif prop == "hostname":
+                if decoded:
+                    return slice_component(url, parsed.host).lower().decode('utf-8')
                 return slice_component(url, parsed.host).lower()
 
 
         cls.__getattr__ = _get_attr
 
-        scheme, netloc, path, query, ref = (slice_component(url, parsed.scheme).lower(),
-                                            build_netloc(url, parsed),
-                                            slice_component(url, parsed.path),
-                                            slice_component(url, parsed.query),
-                                            slice_component(url, parsed.ref))
-        if six.PY2:
+        if decoded == True:
```

Comment:
Is there any reason it's ``if decoded == True:`` instead of ``if decoded:``?

Reply:
No, not at all. I will change it to `decoded:` 😄